bazel/repositories.bzl

```diff
@@ -19,18 +19,29 @@ def proxy_wasm_cpp_sdk_repositories():
     maybe(
         http_archive,
         name = "emsdk",
-        sha256 = "1ca0ff918d476c55707bb99bc0452be28ac5fb8f22a9260a8aae8a38d1bc0e27",
-        # v3.1.7 with Bazel fixes
-        strip_prefix = "emsdk-0ea8f8a8707070e9a7c83fbb4a3065683bcf1799/bazel",
-        url = "https://github.com/emscripten-core/emsdk/archive/0ea8f8a8707070e9a7c83fbb4a3065683bcf1799.tar.gz",
+        sha256 = "52763f556d08ba0c0cde1840102e1e5fcf828b98924c4e77f629ad1d7d400933",
+        # v3.1.44 with one Bazel platforms fix
+        strip_prefix = "emsdk-ef2a8e929d5337755e9b1d1e1d4ad859dc694eee/bazel",
+        url = "https://github.com/emscripten-core/emsdk/archive/ef2a8e929d5337755e9b1d1e1d4ad859dc694eee.tar.gz",
         patches = ["@proxy_wasm_cpp_sdk//bazel:emsdk.patch"],
         patch_args = ["-p2"],
     )
 
     maybe(
         http_archive,
         name = "com_google_protobuf",
-        sha256 = "77ad26d3f65222fd96ccc18b055632b0bfedf295cb748b712a98ba1ac0b704b2",
-        strip_prefix = "protobuf-3.17.3",
-        url = "https://github.com/protocolbuffers/protobuf/releases/download/v3.17.3/protobuf-all-3.17.3.tar.gz",
+        sha256 = "a700a49470d301f1190a487a923b5095bf60f08f4ae4cac9f5f7c36883d17971",
+        strip_prefix = "protobuf-23.4",
+        url = "https://github.com/protocolbuffers/protobuf/releases/download/v23.4/protobuf-23.4.tar.gz",
+    )
+
+    # Update protobuf dependency to fix unused function CodepointSkipBackwards.
+    # https://github.com/protocolbuffers/utf8_range/commit/d863bc33e15cba6d873c878dcca9e6fe52b2f8cb
+    # TODO(martijneken): Remove this when protobuf picks up the fix.
+    maybe(
+        http_archive,
+        name = "utf8_range",
+        sha256 = "568988b5f7261ca181468dba38849fabf59dd9200fb2ed4b2823da187ef84d8c",
+        strip_prefix = "utf8_range-d863bc33e15cba6d873c878dcca9e6fe52b2f8cb",
+        url = "https://github.com/protocolbuffers/utf8_range/archive/d863bc33e15cba6d873c878dcca9e6fe52b2f8cb.zip",
```

Comment:
Note that because C++ SDK is a dependency of C++ Host and projects depending on it, you're going to have to add those dependencies to all of them, otherwise you're breaking them (hence the ask to open draft PRs).

The root cause is that C++ Host unnecessarily depends on C++ SDK because of the headers with ABI definitions and NullVM, but I've been saying for a long time that we should just copy them and develop separately going forward at the cost of some duplication.

Reply:
C++ Host pulls in these repositories: https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/b7e690703c7f26707438a2f1ebd7c197bc8f0296/WORKSPACE#L13

I think you're saying the upstream dependencies do not?

Reply:
Reverted.